source/slang/slang-check-decl.cpp

```diff
@@ -6985,9 +7045,41 @@ namespace Slang
         return "PrimalSubstitute";
     }
 
-    List<Expr*> getImaginaryArgsToFunc(ASTBuilder* astBuilder, FunctionDeclBase* func, SourceLoc loc)
+    // TODO: Can probably use the existing code in lower-to-ir.cpp by moving it somewhere
+    // common.. 
+    //
+    ParameterDirection getParamDirectionFromDecl(ParamDecl* paramDecl)
```

Comment:
Let's not duplicate code, at the very least just declare the function and use it here.